openbas-model/src/main/java/io/openbas/helper/InjectModelHelper.java

```diff
@@ -28,47 +32,89 @@ public static boolean isReady(
+    ArrayNode injectContractFields;
+
+    try {
+      injectContractFields =
+          (ArrayNode)
+              mapper
+                  .readValue(injectorContract.getContent(), ObjectNode.class)
+                  .get(CONTACT_CONTENT_FIELDS);
+    } catch (JsonProcessingException e) {
+      throw new RuntimeException("Error parsing injector contract content", e);
     }
-    AtomicBoolean ready = new AtomicBoolean(true);
+
     ObjectNode contractContent = injectorContract.getConvertedContent();
     List<JsonNode> contractMandatoryFields =
-        StreamSupport.stream(contractContent.get("fields").spliterator(), false)
+        StreamSupport.stream(contractContent.get(CONTACT_CONTENT_FIELDS).spliterator(), false)
             .filter(
-                contractElement ->
-                    (contractElement.get("key").asText().equals("assets")
-                        || contractElement.get("mandatory").asBoolean()
-                        || (contractElement.get("mandatoryGroups") != null
-                            && contractElement.get("mandatoryGroups").asBoolean())))
+                field -> {
+                  String key = field.get(CONTACT_ELEMENT_CONTENT_KEY).asText();
+                  boolean isMandatory = field.get(CONTACT_ELEMENT_CONTENT_MANDATORY).asBoolean();
+                  boolean isMandatoryGroup =
+                      field.hasNonNull(CONTACT_ELEMENT_CONTENT_MANDATORY_GROUPS)
+                          && field.get(CONTACT_ELEMENT_CONTENT_MANDATORY_GROUPS).asBoolean();
+                  return key.equals(CONTACT_ELEMENT_CONTENT_KEY_ASSETS)
+                      || isMandatory
+                      || isMandatoryGroup;
+                })
             .toList();
-    if (!contractMandatoryFields.isEmpty()) {
-      contractMandatoryFields.forEach(
-          jsonField -> {
-            String key = jsonField.get("key").asText();
-            if (key.equals("teams")) {
-              if (teams.isEmpty() && !allTeams) {
-                ready.set(false);
-              }
-            } else if (key.equals("assets")) {
-              if (assets.isEmpty() && assetGroups.isEmpty()) {
-                ready.set(false);
-              }
-            } else if ((jsonField.get("type").asText().equals("text")
-                    || jsonField.get("type").asText().equals("textarea"))
-                && content.get(key) == null) {
-              ready.set(false);
-            } else if ((jsonField.get("type").asText().equals("text")
-                    || jsonField.get("type").asText().equals("textarea"))
-                && content.get(key).asText().isEmpty()) {
-              ready.set(false);
-            }
-          });
+
+    AtomicBoolean isReady = new AtomicBoolean(true);
+    for (JsonNode jsonField : contractMandatoryFields) {
+      String key = jsonField.get(CONTACT_ELEMENT_CONTENT_KEY).asText();
+
+      switch (key) {
+        case CONTACT_ELEMENT_CONTENT_KEY_TEAMS -> {
```

Comment:
Is there a reason you used lambdas here instead of a normal block ? It would avoid the use of an atomic boolean

Reply:
No reason, I'm going to change it.